nixos/modules/services/web-apps/weblate.nix

```diff
@@ -0,0 +1,398 @@
+        example = "weblate.example.org";
+        type = lib.types.str;
+      };
+
+      djangoSecretKeyFile = lib.mkOption {
+        description = ''
+          Location of the Django secret key.
+
+          This should be a string, not a nix path, since nix paths are copied into the world-readable nix store.
+        '';
+        type = lib.types.path;
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Text to append to <filename>settings.py</filename> Weblate config file.
+        '';
+      };
+
+      smtp = {
+        user = lib.mkOption {
+          description = "SMTP login name.";
+          example = "weblate@weblate.example.org";
+          type = lib.types.str;
+        };
+
+        host = lib.mkOption {
+          description = "SMTP host used when sending emails to users.";
+          type = lib.types.str;
+          default = "127.0.0.1";
+        };
+
+        createLocally = lib.mkOption {
+          description = "Configure local Postfix SMTP server for Weblate.";
+          type = lib.types.bool;
+          default = true;
+        };
+        passwordFile = lib.mkOption {
+          description = ''
+            Location of a file containing the SMTP password.
+
+            This should be a string, not a nix path, since nix paths are copied into the world-readable nix store.
+          '';
+          type = lib.types.path;
+        };
+      };
+
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = cfg.smtp.createLocally -> cfg.smtp.host == "127.0.0.1";
+        message = ''services.weblate.smtp.host should be "127.0.0.1" if you want to to use services.weblate.smtp.createLocally.'';
+      }
+    ];
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.localDomain}" = {
+
+        forceSSL = true;
+        enableACME = true;
+
+        locations = {
+          "= /favicon.ico".alias = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/static/favicon.ico";
+          "/static/".alias = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/static/";
+          "/media/".alias = "/var/lib/weblate/media/";
+          "/".extraConfig = ''
+            # Needed for long running operations in admin interface
+            uwsgi_read_timeout 3600;
+            # Adjust based to uwsgi configuration:
+            uwsgi_pass unix:///run/weblate.socket;
+            # uwsgi_pass 127.0.0.1:8080;
+          '';
+        };
+
+      };
+    };
+
+    systemd.services.weblate-postgresql-setup = {
+      description = "Weblate PostgreSQL setup";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "postgresql.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        User = "postgres";
+        Group = "postgres";
+        ExecStart = ''
+          ${pkgs.postgresql}/bin/psql weblate -c "CREATE EXTENSION IF NOT EXISTS pg_trgm"
```

Comment:
Yes, it is enough to add `CREATE` in `ensurePermissions` and run this under the weblate user.

Reply:
(Do note that it may provide access to the postgres user anyway because if someone can do arbitrary `CREATE EXTENSION`, they may be able to load their own ext and do bad stuff, the `postgres` solution is not that bad.)

Reply:
Didn't look further into it, but if the situation is like the two of you described I'd rather execute known code once with higher privileges, than to give the weblate user more privileges for potentially malicious code?

Reply:
Yes, I think letting as-is, is better.